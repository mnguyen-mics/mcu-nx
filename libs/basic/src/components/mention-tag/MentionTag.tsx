import React from 'react';
import { Tag, Tooltip } from 'antd';
import { InfoCircleFilled } from '@ant-design/icons';

export type Mention = 'ALPHA' | 'BETA';

interface State {}

export type MentionTagProps = {
  mention: Mention;
  tooltip?: string;
  className?: string;
  customContent?: string;
};

class MentionTag extends React.Component<MentionTagProps, State> {
  constructor(props: MentionTagProps) {
    super(props);
    this.state = {};
  }

  getMentionClass = (mention: Mention) => {
    const { className } = this.props;
    return `${className ? className : ''} mcs-mentionTag mcs-mentionTag_${mention
      .toString()
      .toLowerCase()}MentionTag`;
  };

  render() {
    const { mention, tooltip, customContent } = this.props;
    if (tooltip) {
      return (
        <Tooltip className='mcs-mentionTag_block' title={tooltip} placement='right'>
          <Tag className={this.getMentionClass(mention)}>{customContent || mention}</Tag>
          <InfoCircleFilled />
        </Tooltip>
      );
    } else {
      return <Tag className={this.getMentionClass(mention)}>{mention}</Tag>;
    }
  }
}

export default MentionTag;
