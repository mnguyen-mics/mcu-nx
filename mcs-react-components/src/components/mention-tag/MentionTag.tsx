import * as React from 'react';
import { Tag, Tooltip } from 'antd';
import { InfoCircleFilled } from '@ant-design/icons';

export type Mention = 'ALPHA' | 'BETA';

interface State {}

export type MentionTagProps = {
  mention: Mention;
  tooltip?: string;
  className?: string;
};

class MentionTag extends React.Component<MentionTagProps, State> {
  constructor(props: MentionTagProps) {
    super(props);
    this.state = {};
  }

  getMentionClass = (mention: Mention) => {
    const { className } = this.props;
    return `${
      className ? className : ''
    } mcs-mentionTag mcs-mentionTag_${mention.toString().toLowerCase()}MentionTag`;
  };

  render() {
    const { mention, tooltip } = this.props;
    if (tooltip) {
      return (
        <Tooltip className='mcs-mentionTag_block' title={tooltip} placement='right'>
          <Tag className={this.getMentionClass(mention)}>{mention}</Tag>
          <InfoCircleFilled />
        </Tooltip>
      );
    } else {
      return <Tag className={this.getMentionClass(mention)}>{mention}</Tag>;
    }
  }
}

export default MentionTag;
