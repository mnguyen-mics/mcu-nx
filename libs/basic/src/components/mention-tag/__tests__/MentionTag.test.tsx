import 'jest';
import React from 'react';
import TestRenderer from 'react-test-renderer';
import MentionTag, { MentionTagProps } from '../MentionTag';

it('renders the MentionTag', () => {
  const props: MentionTagProps = {
    mention: 'ALPHA',
    tooltip: 'tooltip',
  };
  const component = TestRenderer.create(<MentionTag {...props} />);
  const res = component.toJSON();
  expect(res).toMatchSnapshot();
});
