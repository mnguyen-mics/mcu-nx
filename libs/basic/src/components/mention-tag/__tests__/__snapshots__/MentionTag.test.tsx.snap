// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the MentionTag 1`] = `
<span
  className="mcs-mentionTag_block"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <span
    className="ant-tag  mcs-mentionTag mcs-mentionTag_alphaMentionTag"
    style={
      Object {
        "backgroundColor": undefined,
      }
    }
  >
    ALPHA
  </span>
  <span
    aria-label="info-circle"
    className="anticon anticon-info-circle"
    role="img"
  >
    <svg
      aria-hidden="true"
      data-icon="info-circle"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm32 664c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V456c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272zm-32-344a48.01 48.01 0 010-96 48.01 48.01 0 010 96z"
      />
    </svg>
  </span>
</span>
`;
